DB/src/main/java/io/deephaven/db/v2/utils/RuntimeMemory.java

```diff
@@ -88,6 +104,43 @@ public long totalMemory() {
         return lastTotalMemory;
     }
 
+    public long totalCollections() {
+        maybeUpdateValues();
+        return totalCollections;
+    }
+
+    public long getTotalCollectionTimeMs() {
+        maybeUpdateValues();
+        return totalCollectionTimeMs;
+    }
+
+    public static class Sample {
+        public long freeMemory;
+        public long totalMemory;
+        public long totalCollections;
+        public long totalCollectionTimeMs;
+
+        void reset() {
+            freeMemory = totalMemory = totalCollections = totalCollectionTimeMs = 0L;
+        }
+
+        void copy(final Sample s) {
+            freeMemory = s.freeMemory;
+            totalMemory = s.totalMemory;
+            totalCollections = s.totalCollections;
+            totalCollectionTimeMs = s.totalCollectionTimeMs;
+        }
+
+    }
+
+    public void read(Sample buf) {
```

Comment:
I'm pretty sure we should hold a lock that prevents other calls to `maybeUpdateValues()` for the duration of our copy.

Reply:
All these data members are volatile.  If you are concerned about getting a consistent snapshot, we didn't care before for Free/Total, so why would we care now?

Reply:
Nevermind, removed all those methods and made a truly-snapshotting version of the code.